substrate/frame/revive/src/weights.rs

```diff
@@ -2380,47 +2398,60 @@ impl WeightInfo for () {
-		//  Estimated: `6331 + r * (2280 ±0)`
-		// Minimum execution time: 14_670_000 picoseconds.
-		Weight::from_parts(15_744_806, 6331)
-			// Standard Error: 54_003
-			.saturating_add(Weight::from_parts(47_474_393, 0).saturating_mul(r.into()))
+		//  Measured:  `391 + r * (468 ±0)`
+		//  Estimated: `6331 + r * (2162 ±0)`
+		// Minimum execution time: 15_240_000 picoseconds.
+		Weight::from_parts(16_362_393, 6331)
+			// Standard Error: 49_365
+			.saturating_add(Weight::from_parts(47_403_706, 0).saturating_mul(r.into()))
 			.saturating_add(RocksDbWeight::get().reads(2_u64))
 			.saturating_add(RocksDbWeight::get().reads((2_u64).saturating_mul(r.into())))
 			.saturating_add(RocksDbWeight::get().writes(2_u64))
 			.saturating_add(RocksDbWeight::get().writes((3_u64).saturating_mul(r.into())))
-			.saturating_add(Weight::from_parts(0, 2280).saturating_mul(r.into()))
+			.saturating_add(Weight::from_parts(0, 2162).saturating_mul(r.into()))
 	}
 	/// The range of component `r` is `[0, 10000]`.
 	fn evm_opcode(r: u32, ) -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 1_189_000 picoseconds.
-		Weight::from_parts(1_470_372, 0)
-			// Standard Error: 20
-			.saturating_add(Weight::from_parts(6_508, 0).saturating_mul(r.into()))
+		// Minimum execution time: 1_110_000 picoseconds.
+		Weight::from_parts(1_394_699, 0)
+			// Standard Error: 13
+			.saturating_add(Weight::from_parts(6_350, 0).saturating_mul(r.into()))
 	}
 	/// The range of component `r` is `[0, 10000]`.
 	fn instr(r: u32, ) -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 12_052_000 picoseconds.
-		Weight::from_parts(56_794_418, 0)
-			// Standard Error: 416
-			.saturating_add(Weight::from_parts(116_801, 0).saturating_mul(r.into()))
+		// Minimum execution time: 12_552_000 picoseconds.
+		Weight::from_parts(48_405_721, 0)
+			// Standard Error: 453
+			.saturating_add(Weight::from_parts(118_271, 0).saturating_mul(r.into()))
 	}
-	/// The range of component `r` is `[0, 100000]`.
+	/// The range of component `r` is `[0, 10000]`.
 	fn instr_empty_loop(r: u32, ) -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 3_321_000 picoseconds.
-		Weight::from_parts(8_236_588, 0)
-			// Standard Error: 32
-			.saturating_add(Weight::from_parts(71_473, 0).saturating_mul(r.into()))
+		// Minimum execution time: 3_280_000 picoseconds.
+		Weight::from_parts(2_799_697, 0)
+			// Standard Error: 53
+			.saturating_add(Weight::from_parts(71_826, 0).saturating_mul(r.into()))
+	}
+	/// Storage: `Revive::PristineCode` (r:1 w:0)
+	/// Proof: `Revive::PristineCode` (`max_values`: None, `max_size`: None, mode: `Ignored`)
+	/// The range of component `r` is `[1000, 10000]`.
+	fn extcodecopy(r: u32, ) -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `457 + r * (1 ±0)`
+		//  Estimated: `0`
```

Comment:
Is that correct?

Reply:
Good catch. Everything needs to be measured except if its a benchmark like `*_instr` that does not consume PoV. Reading another contracts code should incur PoV usage.

Reply:
arg nice good catch